### PR TITLE
Add warning - component style is overwritten with animation style.

### PR DESCRIPTION
**Pull request by Latropos**
## Summary

In this PR I want to add a warning to prevent cases described here: https://github.com/software-mansion/react-native-reanimated/issues/2975

 
The problem is that some styles are sometimes not affecting Animated.View that has layout animation. It happens when the style property we pass to Animated.View is also animated inside given layout animation. Then it gets overwritten.

Added warning:
> **Warning**
> [Reanimated] Property "transform" of AnimatedComponent(View) may be overwritten with layout animation. Please create a wrapper with the layout animation you want to apply.

<table>
<tr><th>BEFORE</th><th>AFTER</th></tr>
<tr><td>

https://github.com/software-mansion/react-native-reanimated/assets/56199675/8a02aa3b-fc97-4401-9993-d15f184a0c3f

</td><td>

https://github.com/software-mansion/react-native-reanimated/assets/56199675/84b627a3-43dd-4584-8f69-bb1b724dddf0

</td></tr>

</table>

## Test plan

Try this code:
<details>

```js
import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import Animated, { FadeIn, RollInRight } from 'react-native-reanimated';

export default function App() {
  return (
    <View style={styles.container}>
      <Animated.View
        entering={RollInRight.duration(2000)}
        style={{
          width: 200,
          height: 200,
          backgroundColor: 'steelblue',
          transform: [
            { perspective: 1000 },
            { rotateZ: '10deg' },
            { scale: 0.9 },
            { rotateX: '10deg' },
          ],
        }}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 100,
    backgroundColor: '#fff',
    padding: 8,
  },
});
```

</details>

### CODE REVIEW

File: src/animationBuilder.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    'build' in layoutAnimationOrBuilder &&
+    typeof layoutAnimationOrBuilder.build === 'function';
+
+  if (isAnimationBuilder(layoutAnimationOrBuilder)) {
+    const animatedStyle =
+      layoutAnimationOrBuilder.build()(mockTargetValues).animations;
+
+    const getCommonProperties = (obj1: object, obj2: object) =>
+      Object.keys(obj1).filter((key) =>
+        Object.prototype.hasOwnProperty.call(obj2, key)
```
Comment (piaskowyk) [inline src/animationBuilder.tsx:44]:
```suggestion
        obj2.hasOwnProperty(key)
```

File: src/animationBuilder.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+): LayoutAnimationFunction | Keyframe {
+  const isAnimationBuilder = (
+    value: ILayoutAnimationBuilder | LayoutAnimationFunction | Keyframe
+  ): value is ILayoutAnimationBuilder =>
+    'build' in layoutAnimationOrBuilder &&
+    typeof layoutAnimationOrBuilder.build === 'function';
+
+  if (isAnimationBuilder(layoutAnimationOrBuilder)) {
+    const animatedStyle =
+      layoutAnimationOrBuilder.build()(mockTargetValues).animations;
```
Comment (piaskowyk) [inline src/animationBuilder.tsx:40]: Lets spit it to more than one line for readability purposes
```js
const animationFactory = layoutAnimationOrBuilder.build()
const layoutAnimation = animationFactory(mockTargetValues)
const animatedStyle = layoutAnimation.animations
```

File: src/animationBuilder.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import type {
+  ILayoutAnimationBuilder,
+  LayoutAnimationFunction,
+} from './reanimated2/layoutReanimation';
+import type { StyleProps } from './reanimated2/commonTypes';
+
+const mockTargetValues = {
```
Comment (piaskowyk) [inline src/animationBuilder.tsx:7]: Let's add to this proper type `LayoutAnimationsValues` to keep this mock up to date with type.

**Review by piaskowyk — CHANGES_REQUESTED** [review]

File: src/animationBuilder.tsx (hunk 1 above)
Comment (Latropos) [reply src/animationBuilder.tsx:44]: This is banned in eslint recommended: https://eslint.org/docs/latest/rules/no-prototype-builtins
Do we want to turn this rule off?

Comment (piaskowyk) [reply src/animationBuilder.tsx:44]: Ok, it makes sense. But did you check if it works when one of objects is null or undefined?

Comment (Latropos) [reply src/animationBuilder.tsx:44]: It doesn't work, therefore I always pass an object to `getCommonProperties`:
```js
    const commonProperties = getCommonProperties(animatedStyle, style || {});
```

**Review by piaskowyk — APPROVED** [review]
👏

**Pull request merged**